core/client/src/main/java/alluxio/client/AbstractOutStream.java

```diff
@@ -28,12 +28,20 @@
   // requires checking the codebase for this assumption (using int to denote an offset).
   // See ALLUXIO-1536.
   /** The number of bytes written. */
-  protected int mBytesWritten = 0;
+  protected long mBytesWritten = 0;
 
   /**
    * @return the number of bytes written to this stream
+   * @deprecated use {@link #getTotalBytesWritten()} to avoid integer overflow
    */
+  @Deprecated
   public int getBytesWritten() {
+    // Cast to int to preserve backwards compatibility. In 2.0 we should change this to long so that
+    // it can be correct.
+    return (int) mBytesWritten;
+  }
+
+  public long getTotalBytesWritten() {
```

Comment:
I would prefer to not continue supporting this API, since the user could wrap the stream with something like a `CountingOutputStream` to get the same functionality. Can we continue to use `getBytesWritten` in KV (ie. the size limit of the KV partition is less than 2GB) or wrap the stream? @apc999